src/trio/_core/_parking_lot.py

```diff
@@ -86,6 +87,31 @@
     from ._run import Task
 
 
+GLOBAL_PARKING_LOT_BREAKER: dict[Task, list[ParkingLot]] = {}
+
+
+def add_parking_lot_breaker(task: Task, lot: ParkingLot) -> None:
+    """Register a task as a breaker for a lot. This means that if the task exits without
+    having unparked from the lot, then the lot will break and raise an error for all tasks
+    parked in the lot, as well as any future task that attempt to park in it."""
```

Comment:
Hm, this isn't exactly the behavior we're looking for: it's if it exits without removing itself as a parking lot breaker, rather than if it exits while parked on the lot.

Reply:
Ah right. The latter is specifically for `Lock`.